roles/neutron-common/templates/usr/local/bin/neutron-reset

```diff
@@ -201,9 +250,31 @@ class Namespace(object):
 
 
 class ResetNeutron(object):
-    all_agents = ['neutron-dhcp-agent', 'neutron-l3-agent', 'neutron-metadata-proxy', 'neutron-linuxbridge-agent']
+    all_agents = [
+        'neutron-dhcp-agent',
+        'neutron-l3-agent',
+        'neutron-metadata-agent',
+        'neutron-linuxbridge-agent'
+    ]
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--device', help='device to be reset. eg. vxlan')
+    args = parser.parse_args()
 
     def run(self):
+
+        if self.args.device and 'vxlan' in self.args.device:
```

Comment:
Since this is a completely separate routine, can you move this, and the default case (lines 276-326) into separate private methods and dispatch to one of them from run().